src/commons/big_query/big_query_table_metadata.py

```diff
@@ -32,10 +32,19 @@ def get_table_id_with_partition_id(table_id, partition_id):
 
     @staticmethod
     def get_table_by_reference(reference):
```

Comment:
Please refactor the other places where get_table_by_reference is used so it uses your new get_table_by_big_query_table method. If we will not do such a small refactorings during actual work, we will never find time to refactor it

Reply:
these are completely different factory methods, creating BigQueryTableMetadata from two different things:
- TableReference - which is our internal representation of BigQuery Table (eg. recognising partitions)
- BigQueryTable - which is 1:1 representation of how BigQuery api treats table (api use one field tableId for passing tableId and potentially partitionId after '$') 

We were using both representations in different context and although that I agree that we ideally want to use one Table abstractions (instead of 2), I don't think that should be part of this story/pull rq. Especially that requested changes is more about throwing away one of Table representations and using another in every another place, rather than simply using newly created factory method.

Reply:
Will discuss it offline